pkgs/development/tools/analysis/valgrind/default.nix

```diff
@@ -18,7 +18,15 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  patches = stdenv.lib.optionals (stdenv.isDarwin) [ ./valgrind-bzero.patch ];
+  preConfigure = stdenv.lib.optionalString stdenv.isDarwin (
+    let OSRELEASE = ''
+      $(awk -F '"' '/#define OSRELEASE/{ print $2 }' \
+      <${xnu}/Library/Frameworks/Kernel.framework/Headers/libkern/version.h)'';
```

Comment:
we should add something for this to the xnu drv

Reply:
I've been thinking about that, since we actually already encode the macOS version. Sadly those don't easily transpose to the kernel version. Additionally there's only valgrind, IOKit, libpthread and network_cmds that depend directly on `xnu`(as far as I can tell.), so it might not be worth the effort.

Reply:
Yeah, first I was thinking about an attribute on xnu, but that could easily get out of sync. Adding a it to [nix-support](https://github.com/LnL7/nixpkgs/commit/2af109b281bf60a433f6e4ed8b30be823efdbbc3) is the nicest alterrnative I could come up with, but that would be a mass-rebuild.

Reply:
This also seems like a fine solution. 
Just to get an even broader view on the issue I'd like to mention that there also was an idea to replace `uname` itself: https://github.com/NixOS/nixpkgs/pull/25240